torch/csrc/jit/passes/onnx.cpp

```diff
@@ -249,13 +271,33 @@ void NodeToONNX(
         //
         // If onnx shape inference is turned on, the new outputs will have
         // types inferred, and they will be merged with the old types.
-        outputs[i]->setType(MergeInferredType(old->type(), outputs[i]->type()));
+        if (enable_const_folding &&
+            outputs[i]->node()->kind() != c10::onnx::Constant &&
+            ConstantValueMap::HasValue(outputs[i]->debugName())) {
+          // Perform const folding if the node output value is in ConstantValueMap.
+          auto value =
+              ConstantValueMap::GetValue(outputs[i]->debugName()).value();
+          Node* const_node = new_block->owningGraph()->create(c10::onnx::Constant);
+          const_node->t_(attr::value, value);
+          const_node->output()->setType(TensorType::create(value));
+          // const_node->output()->setType(MergeInferredType(old->type(), outputs[i]->type()));
+
+          // Copy over source location and scope information to all nodes
+          // created by the symbolic
+          const_node->output()->node()->setSourceRange(node->sourceRange());
+          const_node->output()->node()->setScope(node->scope());
+          new_block->owningGraph()->appendNode(const_node);
+          ONNXShapeTypeInference(const_node, params_dict, opset_version, static_input_shape);
```

Comment:
I see this is the source of introducing `params_dict` and `opset_version`. Can this be avoided by setting tensortype directly using api tensortype infer from tensor `value`?

Reply:
We don't need pass `params_dict` and `opset_version`.
We create a const node here, in this case, `params_dict={}`. And we get `opset_version` through pybind.
The purpose of `ONNXShapeTypeInference` is used to update record in ConstantValueMap. For each generated ONNX operator (like const here), we need update ConstantValueMap.